src/System.Drawing.Common/tests/BitmapTests.cs

```diff
@@ -0,0 +1,1531 @@
+        {
+            var bitmap = new Bitmap(1, 1);
+            BitmapData data = bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat);
+            data.PixelFormat = format;
+
+            bitmap.UnlockBits(data);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void UnlockBits_NullBitmapData_ThrowsArgumentException()
+        {
+            var bitmap = new Bitmap(1, 1);
+            Assert.Throws<ArgumentException>(null, () => bitmap.UnlockBits(null));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void UnlockBits_NotLocked_ThrowsExternalException()
+        {
+            var bitmap = new Bitmap(1, 1);
+            Assert.Throws<ExternalException>(() => bitmap.UnlockBits(new BitmapData()));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void UnlockBits_AlreadyUnlocked_ThrowsExternalException()
+        {
+            var bitmap = new Bitmap(1, 1);
+            BitmapData data = bitmap.LockBits(new Rectangle(0, 0, 1, 1), ImageLockMode.ReadOnly, bitmap.PixelFormat);
+            bitmap.UnlockBits(data);
+
+            Assert.Throws<ExternalException>(() => bitmap.UnlockBits(new BitmapData()));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void UnlockBits_Disposed_ThrowsArgumentException()
+        {
+            var bitmap = new Bitmap(1, 1);
+            bitmap.Dispose();
+            Assert.Throws<ArgumentException>(null, () => bitmap.UnlockBits(new BitmapData()));
+        }
+
+        public static IEnumerable<object[]> Serialize_TestData()
+        {
+            yield return new object[] { new Bitmap(10, 10), ImageFormat.Png };
+            yield return new object[] { new Bitmap(Helpers.GetTestBitmapPath("16x16_one_entry_4bit.ico")), ImageFormat.Png };
+            yield return new object[] { new Bitmap(Helpers.GetTestBitmapPath("173x183_indexed_8bit.bmp")), ImageFormat.Bmp };
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(Serialize_TestData))]
+        public void Serialize_Deserialize_Roundtrips(Bitmap bitmap, ImageFormat expectedFormat)
+        {
+            using (var memoryStream = new MemoryStream())
+            {
+                var formatter = new BinaryFormatter();
+                formatter.Serialize(memoryStream, bitmap);
+                memoryStream.Position = 0;
+
+                Bitmap deserializedBitmap = (Bitmap)formatter.Deserialize(memoryStream);
+                Assert.Equal(bitmap.Size, deserializedBitmap.Size);
+                Assert.Equal(
```

Comment:
I've updated it in some places. In others, it can't go, such as in the following code, as the index depends on the size
```cs
int n = 0;
Assert.Equal(c.B, data[n++]);
Assert.Equal(c.G, data[n++]);
Assert.Equal(c.R, data[n++]);
if (size % 4 == 0)
{
    if (format == PixelFormat.Format32bppRgb)
    {
        Assert.Equal(128, data[n++]);
    }
    else
    {
        Assert.Equal(c.A, data[n++]);
    }
}
Assert.Equal(d.B, data[n++]);
Assert.Equal(d.G, data[n++]);
Assert.Equal(d.R, data[n++]);
if (size % 4 == 0)
{
    if (format == PixelFormat.Format32bppRgb)
    {
        Assert.Equal(192, data[n++]);
    }
    else
    {
        Assert.Equal(d.A, data[n++]);
    }
}
```